### ISSUE TITLE
Sklearn PCA vs Dask ML PCA

### ISSUE THREAD

**Issue by indrajitsg** [dask/dask-ml#440]
I am trying to compare the results of running PCA (sklearn vs dask). The idea is if this works out, I can use it for large matrices in a distributed environment. I am running the following code for comparison:

```
import numpy as np
import dask.array as da
from dask import compute
from sklearn.decomposition import PCA as skPCA
from dask_ml.decomposition import PCA as daPCA

# Generate random array
np.random.seed(12345)
B = np.random.rand(1000, 1000)

# Create dask array
B_da = da.from_array(B, chunks= B.shape)

# Sklearn PCA with 5 components
sk_pca = skPCA(n_components=5, random_state=6789)
B_pca = sk_pca.fit_transform(B)
print(B_pca)

# Dask PCA with 5 components
da_pca = daPCA(n_components=5, random_state=6789)
B_da_pca = da_pca.fit_transform(B_da)
print(B_da_pca.compute())

```
The results come out very different between sklearn pca and dask-ml pca. Can someone explain why this difference is happening?

**Comment by TomAugspurger** [comment]
dask-ml is using the (randomized) SVD algorithm from http://docs.dask.org/en/latest/array-api.html#dask.array.linalg.svd_compressed. I wouldn't expect it to match scikit-learn exactly, except in trivial cases.

**Comment by indrajitsg** [comment]
Thanks @TomAugspurger for that clarification. Also I noticed that when I try to run this in a distributed environment using dask distributed - by virtue of the following chunk settings:
```
B_da = da.from_array(B, chunks= B.shape)
```
which makes dask treat the entire matrix as a single chunk - the entire processing happens in a single node. Which basically means that it is not running in  a distributed manner. Is there something that I can do to avoid this?

**Comment by TomAugspurger** [comment]
You would need to use a smaller chunk size. Dask Array parallelizes
operations across chunks.

Though note that if your data fits in memory on a single machine then
there's no real reason to use Dask.

**Comment by TomAugspurger** [comment]
Does that answer your questions @indrajitsg?

**Comment by indrajitsg** [comment]
Hi @TomAugspurger - sorry for the delay in response. I was focused on some other experiments, but now I am back to testing Dask PCA and Truncated SVD. Will let you know what I find. Currently my concern is that the results of fitting a Multinomial logit on Dask PCA is giving drastically reduced accuracy metrics as compared to that of Sklearn PCA.

**Comment by TomAugspurger** [comment]
Any news here @indrajitsg?

**Comment by indrajitsg** [comment]
Yes I will close this thread now. Sorry for not missing out this notification for so long.
